pkg/controllers/user/logging/projectlogging.go

```diff
@@ -51,51 +52,58 @@ func registerProjectLogging(cluster *config.UserContext) {
 		projectLoggings:      projectLoggings,
 		serviceAccounts:      cluster.Core.ServiceAccounts(loggingconfig.LoggingNamespace),
 	}
-	projectLoggings.AddClusterScopedLifecycle("project-logging-controller", cluster.ClusterName, lifecycle)
+	projectLoggings.AddClusterScopedHandler("project-logging-controller", cluster.ClusterName, syncer.Sync)
 }
 
-func (c *ProjectLoggingLifecycle) Create(obj *v3.ProjectLogging) (*v3.ProjectLogging, error) {
-	newObj, err := v3.LoggingConditionProvisioned.DoUntilTrue(obj, func() (runtime.Object, error) {
-		return obj, provision(c.namespaces, c.configmaps, c.serviceAccounts, c.clusterRoleBindings, c.daemonsets, c.clusterLister, c.clusterName)
-	})
+func (c *ProjectLoggingSyncer) Sync(key string, obj *v3.ProjectLogging) error {
+	if obj == nil || obj.DeletionTimestamp != nil || utils.GetProjectTarget(obj.Spec) == "none" {
+		isAllDisable, err := utils.CleanResource(c.namespaces, c.clusterLoggingLister, c.projectLoggings.Controller().Lister(), nil, obj)
+		if err != nil {
+			return err
+		}
 
-	return newObj.(*v3.ProjectLogging), err
-}
-func (c *ProjectLoggingLifecycle) Updated(obj *v3.ProjectLogging) (*v3.ProjectLogging, error) {
-	newObj, err := v3.LoggingConditionProvisioned.DoUntilTrue(obj, func() (runtime.Object, error) {
-		return obj, provision(c.namespaces, c.configmaps, c.serviceAccounts, c.clusterRoleBindings, c.daemonsets, c.clusterLister, c.clusterName)
-	})
-	if err != nil {
-		return newObj.(*v3.ProjectLogging), err
+		if obj != nil && !isAllDisable {
+			err = c.createOrUpdateProjectConfigMap(obj.Name)
+		}
+		return err
 	}
 
-	newObj, err = v3.LoggingConditionUpdated.Do(newObj, func() (runtime.Object, error) {
-		return obj, c.createOrUpdateProjectConfigMap("")
-	})
+	original := obj
+	obj = original.DeepCopy()
 
-	if err != nil {
-		return newObj.(*v3.ProjectLogging), err
+	var updateErr error
+	err := c.doSync(obj)
+	if obj != nil && !reflect.DeepEqual(obj, original) {
+		_, updateErr = c.projectLoggings.Update(obj)
 	}
 
-	pl := newObj.(*v3.ProjectLogging)
-	pl.Status.AppliedSpec = pl.Spec
-	return pl, nil
+	if err != nil {
```

Comment:
I notice the same pattern for the second time :) is there a reason why we run update for the case when err might not be nil?